testfixtures/logcapture.py

```diff
@@ -60,7 +80,9 @@ def __init__(self, names=None, install=True, level=1, propagate=None,
         self.attributes = attributes
         self.recursive_check = recursive_check
         self.old = defaultdict(dict)
-        self.clear()
+        self.fail_on_error_log = fail_on_error_log
+        self.records = []  # List[LogRecord]
+        self.asserted = []  # List[bool]
```

Comment:
Leave the call to clear as it was please.

Reply:
I can do that, but I don't think that it is good practice to have members declared outside __init__, It makes the code more difficult to understand. On top of that lint tools don't like it either.